include/drivers/osdp.h

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright (c) 2020 Siddharth Chandrasekaran <siddharth@embedjournal.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _OSDP_H_
+#define _OSDP_H_
+
+#include <zephyr.h>
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Various card formats that a PD can support. This is sent to CP
+ * when a PD must report a card read.
+ */
+enum osdp_card_formats_e {
+	OSDP_CARD_FMT_RAW_UNSPECIFIED,
+	OSDP_CARD_FMT_RAW_WIEGAND,
+	OSDP_CARD_FMT_ASCII,
+	OSDP_CARD_FMT_SENTINEL
+};
+
+/**
+ * @brief Command sent from CP to Control digital output of PD.
+ *
+ * @param output_no 0 = First Output, 1 = Second Output, etc.
+ * @param control_code One of the following:
+ *   0 - NOP – do not alter this output
+ *   1 - set the permanent state to OFF, abort timed operation (if any)
+ *   2 - set the permanent state to ON, abort timed operation (if any)
+ *   3 - set the permanent state to OFF, allow timed operation to complete
+ *   4 - set the permanent state to ON, allow timed operation to complete
+ *   5 - set the temporary state to ON, resume perm state on timeout
+ *   6 - set the temporary state to OFF, resume permanent state on timeout
+ * @param tmr_count Time in units of 100 ms
+ */
+struct osdp_cmd_output {
```

Comment:
If those go over the line they should be `__packed__`

Reply:
No this is filled by code.